### PR TITLE
Put ChainsSpecs in Arc<>

### LINKED ISSUE #3064: Known ChainSpecs should be Arc
### Describe the feature

we already have them as lazy:
https://github.com/paradigmxyz/reth/blob/7cc69d52c6017eeb4348c824a9f5369400e331d1/crates/primitives/src/chain/spec.rs#L16-L16

they should be `Lazy<Arc<ChainSpec>> for convenience and so we don't need an additional instance when we configure `Arc<ChainSpec>`

FYI @joshieDo 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 0xJepsen**
Chain specs are on now in an Atomic Reference Counter Arc<>
Closes #3064

### CODE REVIEW

**Review by rkrasiuk — COMMENTED** [review]
nice, ty

**Review by mattsse — APPROVED** [review]
lgtm, 

pending @joshieDo, since this touches a bit of code changed in #3034

File: crates/staged-sync/src/utils/init.rs
[the remark this answers is not in the document]
Comment (0xJepsen) [reply crates/staged-sync/src/utils/init.rs:1]: Yes latest commit should fix this. Looks like my auto format was on

**Review by gakonst — APPROVED** [review]

**Comment by mattsse** [comment]
this PR did not pass all the checks -.-

**Pull request merged**